jdat_notebooks/niriss_ami_binary/1_niriss_ami_binary.ipynb

```diff
@@ -0,0 +1,645 @@
+    "    yaml_content['Reffiles']['gain'] = './ref_files_non_default/jwst_niriss_gain_general.fits'\n",
+    "    yaml_content['Reffiles']['pixelflat'] = './ref_files_non_default/jwst_niriss_flat_general.fits'\n",
+    "    yaml_content['Reffiles']['superbias'] = './ref_files_non_default/jwst_niriss_superbias_sim.fits'\n",
+    "\n",
+    "    \n",
+    "    if \"jw01093001\" in file:\n",
+    "       yaml_content['Reffiles']['dark'] = './ref_files_non_default/simdarks/dark000001/dark000001_uncal.fits'\n",
+    "    elif \"jw01093002\" in file:\n",
+    "       yaml_content['Reffiles']['dark'] = './ref_files_non_default/simdarks/dark000005/dark000005_uncal.fits'\n",
+    "    \n",
+    "\n",
+    "    modified_file = file.replace('.yaml', '_mod.yaml')\n",
+    "    with io.open(modified_file, 'w') as outfile:\n",
+    "        yaml.dump(yaml_content, outfile, default_flow_style=False)\n",
+    "\n",
+    "\n",
+    "    print(\"Updated yaml files. The jw*_mod.yaml files will be used to create data\")\n",
+    "    \n",
+    "    #create data\n",
+    "    t1 = imaging_simulator.ImgSim()\n",
+    "    t1.paramfile = str(modified_file)\n",
+    "    t1.create()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Useful output products generated by Mirage\n",
+    "\n",
+    "```\n",
+    "- jw01093001001_01101_00001_nis_uncal_pointsources.list and \n",
+    "  jw01093002001_01101_00001_nis_uncal_pointsources.list. The first line of \n",
+    "  jw01093001001_01101_00001_nis_uncal_pointsources.list shows coordinates of AB Dor, the\n",
+    "  pixel coordinates at which the data is simulated, magnitude and total count rate.\n",
+    "   \n",
+    "- jw01093001001_01101_00001_nis_uncal_F480M_NRM_final_seed_image.fits and  \n",
+    "  jw01093002001_01101_00001_nis_uncal_F480M_NRM_final_seed_image.fits\n",
+    "  A seed image is a noiseless image that contains signal only from simulated \n",
+    "  astronomical sources. It can be used for quality checks on the final output data.\n",
+    "\n",
+    "- jw01093001001_01101_00001_nis_uncal.fits and jw01093002001_01101_00001_nis_uncal.fits\n",
+    "  AB Dor and HD37093 raw data\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Examine the seed images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_seed = []\n",
+    "seed_images = sorted(glob.glob('mirage_sim_data/jw*final_seed_image.fits'))\n",
+    "for i, df in enumerate(seed_images):\n",
+    "    seed_im = fits.open(df)\n",
+    "    seed_im.info()\n",
+    "    im = seed_im[1].data    \n",
+    "    print(im.shape)\n",
+    "    data_seed.append(im)\n",
+    "print(data_seed[0].shape, data_seed[1].shape)\n",
+    "\n",
+    "#Temporarily adding the two lines below. \n",
```

Comment:
These lines can be removed if you switch the order of the imports at the top of the notebook. matplotlib statements should be last in that cell, after Mirage.